irrlichtmt: init at 1.9.0mt4; minetest: use irrlichtmt, add deprecation warning aliases for _4

DeeUnderscore opened the pull request:
###### Motivation for this change
* Minetest release 5.5.0: https://github.com/minetest/minetest/releases/tag/5.5.0
* irrlichtmt is a fork of irrlicht by the Minetest project that Minetest switched to in 5.5.0
* Dropping `minetestclient_4` and `minetestserver_4`, since the Minetest 5.0.0 release was in 2019, and stuff saved in 4.x is compatible with versions 5.0.0 and up. The main reason to retain _4 otherwise would be to allow connecting to servers running the old version. Maintaining _4 would require refactoring the derivation, since the build inputs now differ between versions. 

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

iclanzan commented on pkgs/games/minetest/default.nix:
```diff
@@ -35,7 +38,7 @@ let
       "-DENABLE_GETTEXT=1"
       "-DENABLE_SYSTEM_JSONCPP=1"
-      "-DIRRLICHT_INCLUDE_DIR=${irrlicht}/include/irrlicht"
+      "-DIRRLICHT_INCLUDE_DIR=${irrlichtmt}/include/irrlichtmt"
```
Since Minetest 5.5.0 can be compiled with touch support, can we add an option for that? `ENABLE_TOUCH` is the flag according to the README at https://github.com/minetest/minetest.

DeeUnderscore replied:
I added `withTouchSupport`, which also requires rebuilding IrrlichtMT (upstream is not set up to build with touchscreen support by default), but I did test it on my Thinkpad X200 Tablet (which is the one x86 touchscreen device running NixOS I have) and it works with multitouch support and everything.

fpletz commented:
Oh no, I'm so sorry, I haven't seen your PR and did the update myself: https://github.com/NixOS/nixpkgs/pull/158134

Would you be willing to update your PR to add that separate irrlichtmt attribute, the touch support and the release notes?

DeeUnderscore committed:
irrlichtmt: init at 1.9.0mt4

3noch commented:
Would this fix #166742 ?

DeeUnderscore commented:
Probably not, it essentially builds the same thing as before by default.

DeeUnderscore committed:
minetest: use irrlichtmt from Nixpkgs, add optional touch support

Touch support requires rebuilding irrlichtmt with touch support, which
upstream does not enable by default. Building Minetest with touch
support also disables the regular interface, so touch support defaults
to off.

DeeUnderscore committed:
minetest: add aliases and release notes for v4 deprecation

Mindavi commented on pkgs/development/libraries/irrlichtmt/default.nix:
```diff
@@ -0,0 +1,61 @@
+, libGL
+, libX11
+, libXxf86vm
+, withTouchSupport ? false
```
Any reason why this is optional?

DeeUnderscore replied:
Mostly to reflect upstream, which currently does not enable it by default. I think their concern is mostly not wanting to pull in another dependency, which is less of a concern here, but I wanted to err on the side of caution.

Mindavi replied:
Having to build this on a pinephone might not be ideal either 😃. But yeah, I can imagine that you would generally like to follow the advice of upstream. It's just that now, out of the box, touch support won't be working on a touch-enabled device and a (re)build would be needed.

Mindavi commented on pkgs/development/libraries/irrlichtmt/default.nix:
```diff
@@ -0,0 +1,61 @@
+  # https://github.com/minetest/minetest/pull/10729
+  postPatch = lib.optionalString withTouchSupport ''
+    substituteInPlace include/IrrCompileConfig.h \
+      --replace '//#define _IRR_LINUX_X11_XINPUT2_' '#define _IRR_LINUX_X11_XINPUT2_'
```
Could also be a compile flag (-D_IRR...), but this is fine.

DeeUnderscore replied:
Upstream doesn't have a Cmake option for this, and [the build script](https://gist.github.com/TheBrokenRail/d1ef86d115fa8012cff13af350c5fdfd) from [the PR](https://github.com/minetest/minetest/pull/10729) that enabled touch in Minetest pretty much does this.

Mindavi replied:
Yeah, it was just a suggestion. This is fine too.

Mindavi reviewed (approved):
Diff LGTM, didn't test. Consider enabling touch support unconditionally, unless there's a good reason not to.
